pkg/loop/internal/core/services/capability/standard_capability.go

```diff
@@ -0,0 +1,223 @@
+	"github.com/smartcontractkit/chainlink-common/pkg/services"
+	"github.com/smartcontractkit/chainlink-common/pkg/types/core"
+)
+
+type StandardCapability interface {
+	services.Service
+	capabilities.BaseCapability
+	Initialise(ctx context.Context, config string, telemetryService core.TelemetryService, store core.KeyValueStore,
+		capabilityRegistry core.CapabilitiesRegistry, errorLog core.ErrorLog,
+		pipelineRunner core.PipelineRunnerService, relayerSet core.RelayerSet) error
+}
+
+type StandardCapabilityClient struct {
+	*goplugin.PluginClient
+	capabilitiespb.StandardCapabilityClient
+	*baseCapabilityClient
+	*goplugin.ServiceClient
+	*net.BrokerExt
+
+	resources []net.Resource
+}
+
+var _ StandardCapability = (*StandardCapabilityClient)(nil)
+
+func NewStandardCapabilityClient(brokerExt *net.BrokerExt, conn *grpc.ClientConn) *StandardCapabilityClient {
+	return &StandardCapabilityClient{
+		PluginClient:             goplugin.NewPluginClient(brokerExt.Broker, brokerExt.BrokerConfig, conn),
+		ServiceClient:            goplugin.NewServiceClient(brokerExt, conn),
+		StandardCapabilityClient: capabilitiespb.NewStandardCapabilityClient(conn),
+		baseCapabilityClient:     newBaseCapabilityClient(brokerExt, conn),
+		BrokerExt:                brokerExt,
+	}
+}
+
+func (c *StandardCapabilityClient) Initialise(ctx context.Context, config string, telemetryService core.TelemetryService,
+	keyValueStore core.KeyValueStore, capabilitiesRegistry core.CapabilitiesRegistry, errorLog core.ErrorLog,
+	pipelineRunner core.PipelineRunnerService, relayerSet core.RelayerSet) error {
+	telemetryID, telemetryRes, err := c.ServeNew("Telemetry", func(s *grpc.Server) {
+		pb.RegisterTelemetryServer(s, telemetry.NewTelemetryServer(telemetryService))
+	})
+	if err != nil {
+		return fmt.Errorf("failed to serve new telemetry: %w", err)
+	}
+	c.resources = append(c.resources, telemetryRes)
+
+	keyValueStoreID, keyValueStoreRes, err := c.ServeNew("KeyValueStore", func(s *grpc.Server) {
+		pb.RegisterKeyValueStoreServer(s, keyvalue.NewServer(keyValueStore))
+	})
+	if err != nil {
+		return fmt.Errorf("failed to serve new key value store: %w", err)
+	}
+	c.resources = append(c.resources, keyValueStoreRes)
+
+	capabilitiesRegistryID, capabilityRegistryResource, err := c.ServeNew("CapabilitiesRegistry", func(s *grpc.Server) {
+		pb.RegisterCapabilitiesRegistryServer(s, NewCapabilitiesRegistryServer(c.BrokerExt, capabilitiesRegistry))
+	})
+	if err != nil {
+		return fmt.Errorf("failed to serve new key value store: %w", err)
+	}
+	c.resources = append(c.resources, capabilityRegistryResource)
+
+	errorLogID, errorLogRes, err := c.ServeNew("ErrorLog", func(s *grpc.Server) {
+		pb.RegisterErrorLogServer(s, errorlog.NewServer(errorLog))
+	})
+	if err != nil {
+		return fmt.Errorf("failed to serve error log: %w", err)
+	}
+	c.resources = append(c.resources, errorLogRes)
+
+	pipelineRunnerID, pipelineRunnerRes, err := c.ServeNew("PipelineRunner", func(s *grpc.Server) {
```

Comment:
Why do we need a pipeline runner?

Reply:
My understanding is that the standard capability is analogous to a generic plugin, and thus should have the same resources available to it, one of those being pipeline runner.  IIRC I did raise this specific question regarding pipeline runner with Cedric to check (same as for keyvalue store).  As I understand it the idea is that we give the capability/plugin writers as much functionality as possible to avoid them having to write custom implementations that would have to access core code directly.

Reply:
Do we plan to use this for any existing capabilities? If not, is it worth considering keeping the API surface small if there is a way to extend the API in a non-breaking way later?

Reply:
> Do we plan to use this for any existing capabilities? If not, is it worth considering keeping the API surface small if there is a way to extend the API in a non-breaking way later?

Nvm, we actually need it for the Consensus capability.